src/main/java/io/cdap/plugin/http/source/batch/HttpBatchSourceConfig.java

```diff
@@ -29,6 +49,73 @@ protected HttpBatchSourceConfig(String referenceName) {
   @Override
   public void validate(FailureCollector failureCollector) {
     super.validate(failureCollector);
+    validateCredentials(failureCollector);
+  }
+
+  public void validateCredentials(FailureCollector collector) {
+    try {
+      if (getAuthType() == AuthType.OAUTH2) {
+        validateOAuth2Credentials(collector);
+      } else if (getAuthType() == AuthType.BASIC_AUTH) {
+        validateBasicAuthCredentials(collector);
+      }
+    } catch (IOException e) {
+      throw new IllegalStateException("Unable to authenticate the given info", e);
+    }
+  }
+
+  private void validateOAuth2Credentials(FailureCollector collector) throws IOException {
+    try {
+      if (!containsMacro(PROPERTY_CLIENT_ID) && !containsMacro(PROPERTY_CLIENT_SECRET) &&
+        !containsMacro(PROPERTY_TOKEN_URL) && !containsMacro(PROPERTY_REFRESH_TOKEN) &&
+        !containsMacro(PROPERTY_PROXY_PASSWORD) && !containsMacro(PROPERTY_PROXY_USERNAME) &&
+        !containsMacro(PROPERTY_PROXY_URL)) {
+        HttpClientBuilder httpclientBuilder = HttpClients.custom();
+        if (!containsMacro(PROPERTY_PROXY_URL) && !Strings.isNullOrEmpty(getProxyUrl())) {
+          HttpHost proxyHost = HttpHost.create(getProxyUrl());
+          if (!Strings.isNullOrEmpty(getProxyUsername()) && !Strings.isNullOrEmpty(getProxyPassword())
+            && !containsMacro(PROPERTY_PROXY_USERNAME) && !containsMacro(PROPERTY_PROXY_PASSWORD)) {
+            CredentialsProvider credsProvider = new BasicCredentialsProvider();
+            credsProvider.setCredentials(new AuthScope(proxyHost),
+              new UsernamePasswordCredentials(getProxyUsername(), getProxyPassword()));
+            httpclientBuilder.setDefaultCredentialsProvider(credsProvider);
+          }
+          httpclientBuilder.setProxy(proxyHost);
+        }
+        CloseableHttpClient closeableHttpClient = httpclientBuilder.build();
```

Comment:
Please make sure to close the client after use.

Reply:
done